astropy/io/ascii/cparser.pyx

```diff
@@ -152,7 +153,7 @@ cdef class FileString:
         cdef char *ptr = <char *>self.mmap_ptr
         cdef char *tmp
         cdef int line_len
-        cdef int map_len = len(self.mmap)
+        cdef int map_len = <int>len(self.mmap)
```

Comment:
this is wrong, never use int in cython code when not absolutely needed (which is very rare)

mmap is a pointer to memory so it can exceed 32 bit on 64 bit machines, use a wordsized type, like Py_ssize_t, intptr_t, npy_intp, size_t, ...